src/resolve/resolved-dns-server.c

```diff
@@ -828,6 +828,37 @@ void dns_server_flush_cache(DnsServer *s) {
         dns_cache_flush(&scope->cache);
 }
 
+void dns_server_reexamine(DnsServer *s) {
+        assert(s);
+
+        s->max_rtt = 0;
+        s->resend_timeout = DNS_TIMEOUT_MIN_USEC;
+
+        s->verified_feature_level = _DNS_SERVER_FEATURE_LEVEL_INVALID;
+        s->possible_feature_level = DNS_SERVER_FEATURE_LEVEL_BEST;
+
+        s->received_udp_packet_max = DNS_PACKET_UNICAST_SIZE_MAX;
+
+        s->n_failed_udp = 0;
+        s->n_failed_tcp = 0;
```

Comment:
Why not call `dns_server_reset_counters()` here?

I also think `dns_server_reset` would be more in line with the other function names. This function doesn't "examine" anything and I don't think we say "reexamine" anywhere else in the systemd codebase.